### PR TITLE
alert disappear after time

### LINKED ISSUE #56: [FEATURE] Success alert disappear
### Is this a unique feature?

- [X] I have checked "open" AND "closed" issues and this is not a duplicate

### Is your feature request related to a problem/unavailable functionality? Please describe.

Once the user has provided a rating, the success alert should disappear automatically after a few seconds.

### Proposed Solution

Success alert should disappear after some seconds

### Screenshots

Screenshot 2023-12-30 154643

### Do you want to work on this issue?

Yes

### If "yes" to above, please explain how you would technically implement this

I think the setTimeout function solve this.

### PR DESCRIPTION
**Pull request by amanr-dev**
## Fixes Issue
Fixes #56 

## Changes proposed
Hide the Success alert after some seconds.

## Check List (Check all the applicable boxes) 

- [ ] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [ ] This PR does not contain plagiarized content.
- [ ] The title of my pull request is a short description of the requested changes.

## Summary by CodeRabbit

- **Refactor**
	- Improved alert display and navigation handling in the `Page` component using React hooks for state management and side effects.

### CODE REVIEW

File: src/app/page.js [hunk 1]
```diff
@@ -1,16 +1,14 @@
...
 
 import SideNav from "@/components/SideNav";
 import Repos from "@/components/Repos";
 import Activity from "@/components/Activity";
 import Stats from "@/components/Stats";
 import Toast from "@/components/Toast";
-import { useSearchParams } from "next/navigation";
-import { Menu, Transition } from "@headlessui/react";
-import { ChevronUpDownIcon } from "@heroicons/react/24/outline";
-import { classNames } from "@/utils/classNames";
```
Comment (eddiejaoude) [inline src/app/page.js:13]: Not sure why these are removed? The page will break, please test

File: src/app/page.js [hunk 2]
```diff
@@ -1,22 +1,35 @@
...
   const [keyword, setKeyword] = useState("");
+  const [showAlert, setShowAlert] = useState(false);
+
+  useEffect(() => {
+    if (alert) {
+      setShowAlert(true);
+      setTimeout(() => {
+        setShowAlert(false);
+        // remove alert be
+        router.push(`/?done=success&message=Rating saved!`, { scroll: false });
```
[the remark this answers is not in the document]
Comment (eddiejaoude) [reply src/app/page.js:34]: Not sure why this was marked as resolved?

[the remark this answers is not in the document]
Comment (eddiejaoude) [reply src/app/page.js:34]: This does not remove the alert? The comment doesn't seem to support the code?

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
Thank you 👍  I left some inline comments

File: src/app/page.js (hunk 1 above)
Comment (amanr-dev) [reply src/app/page.js]: It was removed by me while fixing the conflict.

**Comment by amanr-dev** [comment]
Added all the removed code

**Comment by eddiejaoude** [comment]
Thank you, please don't marked things as resolved, it makes it difficult to review

**Comment by amanr-dev** [comment]
> Thank you, please don't marked things as resolved, it makes it difficult to review

ok and sorry

File: src/app/page.js [hunk 3]
```diff
@@ -21,10 +21,23 @@ const sortOptions = [
...
   const [sort, setSort] = useState(sortOptions[0].value);
+  const [showAlert, setShowAlert] = useState(false);
+
+  useEffect(() => {
+    if (alert) {
+      setShowAlert(true);
+      setTimeout(() => {
+        setShowAlert(false);
+        // Removed alert because it can re-render the Toast Component
+        router.push(`/?done=success&message=${message}`, { scroll: false });
```
Comment (eddiejaoude) [inline src/app/page.js:37]: As mentioned in a previous inline comment, I think this should be `/`?

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
Looks good 👍 , I just 1 inline comment that gone missed before

File: src/app/page.js (hunk 3 above)
Comment (amanr-dev) [reply src/app/page.js]: changed router path `/?done=success&message=${message}` to `/`

**Comment by amanr-dev** [comment]
Previously, I thought that, its not good to remove all the queries. That's why I only changed `alert`.

File: src/app/page.js (hunk 3 above)
Comment (eddiejaoude) [reply src/app/page.js]: I think this is correct otherwise the alert will keep appearing? Did you test it?

Comment (amanr-dev) [reply src/app/page.js]: Yes, tested it.

Comment (amanr-dev) [reply src/app/page.js]: I tested it many times until I finally found the solution.

Comment (amanr-dev) [reply src/app/page.js]: > changed router path `/?done=success&message=${message}` to `/`
Then I found that this `/?done=success&message=${message}` will be a better way to stop re-rendering.

File: src/app/page.js [hunk 4]
```diff
@@ -21,10 +21,23 @@ const sortOptions = [
 
 export default function Page() {
   const params = useSearchParams();
+  const router = useRouter();
   const alert = params.get("alert");
   const message = params.get("message");
   const [keyword, setKeyword] = useState("");
   const [sort, setSort] = useState(sortOptions[0].value);
+  const [showAlert, setShowAlert] = useState(false);
```
Comment (eddiejaoude) [inline src/app/page.js:29]: With the new changes of the redirect not having the params, this is no longer required

```suggestion
```

File: src/app/page.js [hunk 5]
```diff
@@ -21,10 +21,23 @@ const sortOptions = [
...
+
+  useEffect(() => {
+    if (alert) {
+      setShowAlert(true);
```
Comment (eddiejaoude) [inline src/app/page.js:33]: This is now also no longer needed

```suggestion
```

File: src/app/page.js [hunk 6]
```diff
@@ -21,10 +21,23 @@ const sortOptions = [
...
+      setTimeout(() => {
+        setShowAlert(false);
```
Comment (eddiejaoude) [inline src/app/page.js:35]: This can go also

```suggestion
```

File: src/app/page.js [hunk 7]
```diff
@@ -21,10 +21,23 @@ const sortOptions = [
...
+        // Removed alert because it can re-render the Toast Component
```
Comment (eddiejaoude) [inline src/app/page.js:36]:
```suggestion
```

File: src/app/page.js [hunk 8]
```diff
@@ -82,7 +95,7 @@ export default function Page() {
           </aside>
         </>
       </SideNav>
-      {alert && <Toast type={alert} message={message} />}
+      {showAlert && <Toast type={alert} message={message} />}
```
Comment (eddiejaoude) [inline src/app/page.js:98]: Then this becomes `alert`

```suggestion
      {alert && <Toast type={alert} message={message} />}
```

**Review by eddiejaoude — COMMENTED** [review]
This can now be simplified

File: src/app/page.js [hunk 9]
```diff
@@ -21,11 +21,20 @@ const sortOptions = [
...
   const alert = params.get("alert");
   const message = params.get("message");
   const [keyword, setKeyword] = useState("");
   const [sort, setSort] = useState(sortOptions[0].value);
 
+  useEffect(() => {
+    if (alert) {
+      setTimeout(() => {
+        router.push("/", { scroll: false });
+      }, 4000);
```
Comment (eddiejaoude) [inline src/app/page.js:34]: This now can be simplified also

```suggestion
      setTimeout(() => router.push("/", { scroll: false }), 4000);
```

**Review by eddiejaoude — APPROVED** [review]
Thank you for contributing 👍

**Pull request merged**